Description: Do not link libcups.so and libcupsimage.so against unnecessary libraries.  This avoids unnecessary package dependencies for both libcups, as well as for packages using cups-config.
Author: Martin Pitt <mpitt@debian.org>
Bug-Debian: http://bugs.debian.org/438067

@DPATCH@
Index: cups-1.4.6/Makedefs.in
===================================================================
--- cups-1.4.6.orig/Makedefs.in	2011-07-14 13:02:00.959772015 +0200
+++ cups-1.4.6/Makedefs.in	2011-07-14 13:02:32.379772550 +0200
@@ -132,7 +132,7 @@
 DBUS_NOTIFIERLIBS =	@DBUS_NOTIFIERLIBS@
 DNSSD_BACKEND	=	@DNSSD_BACKEND@
 DSOFLAGS	=	-L../cups @DSOFLAGS@
-DSOLIBS		=	@DSOLIBS@ $(COMMONLIBS)
+DSOLIBS		=	@DSOLIBS@
 DNSSDLIBS	=	@DNSSDLIBS@
 FONTS		=	@FONTS@
 IMGLIBS		=	@IMGLIBS@
@@ -144,9 +144,9 @@
 LEGACY_BACKENDS	=	@LEGACY_BACKENDS@
 LIBCUPSORDER	=	@LIBCUPSORDER@
 LIBCUPSIMAGEORDER =	@LIBCUPSIMAGEORDER@
-LINKCUPS	=	@LINKCUPS@ $(SSLLIBS) $(DNSSDLIBS)
+LINKCUPS	=	@LINKCUPS@ $(DNSSDLIBS)
 LINKCUPSIMAGE	=	@LINKCUPSIMAGE@
-LIBS		=	$(LINKCUPS) $(COMMONLIBS)
+LIBS		=	$(LINKCUPS) $(SSLLIBS) $(COMMONLIBS)
 OPTIM		=	@OPTIM@
 OPTIONS		=
 PAMLIBS		=	@PAMLIBS@
Index: cups-1.4.6/cups/Makefile
===================================================================
--- cups-1.4.6.orig/cups/Makefile	2011-07-14 13:02:00.979772014 +0200
+++ cups-1.4.6/cups/Makefile	2011-07-14 13:02:32.379772550 +0200
@@ -294,7 +294,7 @@
 libcups.so.2 libcups.sl.2:	$(LIBOBJS)
 	echo Linking $@...
 	$(DSO) $(ARCHFLAGS) $(DSOFLAGS) -o $@ $(LIBOBJS) $(LIBGSSAPI) \
-		$(SSLLIBS) $(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
+		$(SSLLIBS) $(DNSSDLIBS) -lpthread $(LIBZ)
 	$(RM) `basename $@ .2`
 	$(LN) $@ `basename $@ .2`
 
@@ -307,7 +307,7 @@
 	echo Linking 32-bit $@...
 	-mkdir 32bit
 	$(DSO) $(ARCH32FLAGS) $(DSO32FLAGS) -o $@ $(LIB32OBJS) $(LIBGSSAPI) \
-		$(DNSSDLIBS) $(SSLLIBS) $(COMMONLIBS) $(LIBZ)
+		$(DNSSDLIBS) $(SSLLIBS) -lpthread $(LIBZ)
 	$(RM) 32bit/libcups.so
 	$(LN) libcups.so.2 32bit/libcups.so
 
@@ -320,7 +320,7 @@
 	echo Linking 64-bit $@...
 	-mkdir 64bit
 	$(DSO) $(ARCH64FLAGS) $(DSO64FLAGS) -o $@ $(LIB64OBJS) $(LIBGSSAPI) \
-		$(DNSSDLIBS) $(SSLLIBS) $(COMMONLIBS) $(LIBZ)
+		$(DNSSDLIBS) $(SSLLIBS) -lpthread $(LIBZ)
 	$(RM) 64bit/libcups.so
 	$(LN) libcups.so.2 64bit/libcups.so
 
@@ -338,7 +338,7 @@
 		-exported_symbols_list libcups.exp \
 		-sectorder __TEXT __text $(LIBCUPSORDER) \
 		$(LIBOBJS) $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
-		$(COMMONLIBS) $(LIBZ)
+		-lpthread $(LIBZ)
 	$(RM) libcups.dylib
 	$(LN) $@ libcups.dylib
 
@@ -351,7 +351,7 @@
 	echo Creating $@...
 	$(DSO) $(DSOFLAGS) -Wl,-bexport:libcups_s.exp -o libcups_s.o \
 		$(LIBOBJS) $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
-		$(COMMONLIBS) $(LIBZ) -lm
+		-lpthread $(LIBZ)
 	$(RM) $@
 	$(AR) $(ARFLAGS) $@ libcups_s.o
 
@@ -364,7 +364,7 @@
 	echo Linking $@...
 	$(CC) $(ARCHFLAGS) $(DSOFLAGS) -o $@ $(LIBOBJS:.o=.lo) \
 		-rpath $(LIBDIR) -version-info 2:8 $(LIBGSSAPI) $(SSLLIBS) \
-		$(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
+		$(DNSSDLIBS) -lpthread $(LIBZ)
 
 
 #
